src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -179,9 +179,20 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeIsReadOnlyAttribute(this));
             }
 
-            if (this.IsParams && this.ContainingSymbol is SynthesizedDelegateInvokeMethod)
+            if (this.ContainingSymbol is SynthesizedDelegateInvokeMethod or SynthesizedClosureMethod)
             {
-                AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_ParamArrayAttribute__ctor));
+                if (this.IsParams)
+                {
+                    AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_ParamArrayAttribute__ctor));
+                }
```

Comment:
📝 The outer condition now also admits lambda/local function implementation methods, but their parameters are never marked as `params`, so nothing is changed for that code path. And if those parameters were marked as `params`, we would probably want to emit the attribute anyway. Alternatively, I can of course lift `if (this.IsParams)`.

Reply:
> but their parameters are never marked as params, so nothing is changed for that code path. And if those parameters were marked as params, we would probably want to emit the attribute anyway.

I think it would be better to not make assumptions like that and precisely target scenario that we want to enable today.

Reply:
OK, thanks, changed.